### ISSUE TITLE
Breadcrumbs ignore "number-sections: false" setting in book project web site

### ISSUE THREAD

**Issue by blacksqr** [quarto-dev/quarto-cli#6184]
### Bug description

When specifying `number-sections: false` in a book project, chapter numbers still appear in the breadcrumb navigation links at the top of chapter pages in the web site.

### Steps to reproduce

Create a default book project:

```
$ quarto create-project mybook --type book
$ quarto preview mybook
```

Edit _quarto.yml as follows (add a `part:` option so pages have breadcrumbs, specify `number-sections: false`):

```
project:
  type: book

book:
  title: "mybook"
  author: "Norah Jones"
  date: "7/11/2023"
  chapters:
    - index.qmd
    - part: "Part One"
      chapters:
      - intro.qmd
      - summary.qmd
    - references.qmd

bibliography: references.bib

format:
  html:
    theme: cosmo
    number-sections: false
  pdf:
    documentclass: scrreprt
```

### Expected behavior

The breadcrumbs for the rendered intro.html page should appear as:

`Part One > Introduction`

### Actual behavior

The breadcrumbs appear as:

`Part One > 1 Introduction`

### Your environment

- Quarto version 1.4.217
- OS: Kubuntu 20.04

**Comment by blacksqr** [comment]
I reported this as happening in 1.4.217.  It doesn't appear to have been fixed in subsequent commits.

**Comment by mcanouil** [comment]
@blacksqr it's very likely to be a typo in the version number indicated, did you try to update your version before posting to see if the fix is working?

**Comment by blacksqr** [comment]
The commit you mention is included in 1.4.217 and addresses a separate issue than the one I report here.  I've looked at the subsequent commits and I didn't find any changes that appear to address this issue.
